api_app/analyzers_manager/observable_analyzers/onionscan.py

```diff
@@ -0,0 +1,82 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import json
+import logging
+import subprocess
+from shutil import which
+
+# celery exceptions
+from celery.exceptions import SoftTimeLimitExceeded
+
+from api_app.analyzers_manager import classes
+from api_app.exceptions import AnalyzerRunException
+
+# test mocks
+from tests.mock_utils import MockPopen, patch
+
+logger = logging.getLogger(__name__)
+
+
+class OnionScan(classes.ObservableAnalyzer):
+    """
+    check a .onion domain for privacy misconfigurations and info leaks.
+    """
+
+    onionscan_binary: str = "/opt/deploy/bundled/onionscan"
+    _verbose: bool = True
+    _tor_proxy_address: str = None
+
+    def set_params(self, params):
+        self._verbose = params.get("verbose", True)
+        self._tor_proxy_address = params.get("torProxyAddress", None)
+
+    def run(self):
+        """
+        Run Onionscan against target onion domain
+        """
+        # Check for onionscan binary in path/pwd.
+        if which("onionscan"):
+            self.onionscan_binary = "onionscan"
```

Comment:
is this really required ? we are not installing onionscan so this will never be true.

Reply:
Yeah, that's true. As we are using a docker container and scripts I think its better to remove it.